src/vm/interpreter.cpp

```diff
@@ -1431,8 +1431,8 @@ CorJitResult Interpreter::GenerateInterpreterStub(CEEInfo* comp,
         sl.X86EmitPopReg(kEBP);
         sl.X86EmitReturn(static_cast<WORD>(argState.callerArgStackSlots * sizeof(void*)));
 #elif defined(_AMD64_)
-        // EDX has "ilArgs" i.e., just the point where registers have been homed.
-        sl.X86EmitIndexLeaRSP(kEDX, static_cast<X86Reg>(kESP_Unsafe), 8);
+        // Pass "ilArgs", i.e. just the point where registers have been homed, as 2nd arg
+        sl.X86EmitIndexLeaRSP(kRSI, static_cast<X86Reg>(kESP_Unsafe), 8);
```

Comment:
This is not correct, kRSI is the 2nd argument on Unix only. On Windows, it is the kEDX that was there before. The best way would be to introduce ARGUMENT_kREG1 / ARGUMENT_kREG2 defines and put them to cgencpu.h for amd64 next to the THIS_REG definition.